### ISSUE TITLE
Remove compiler option to suppress warnings for using experimental unsigned types

### ISSUE THREAD

**Issue by thunderbiscuit** [bitcoindevkit/bdk-kotlin#11]
The following block is no longer necessary as of Kotlin 1.5 (we're currently using 1.6.10):
```kotlin
allprojects {
    tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().configureEach {
        kotlinOptions.freeCompilerArgs += "-opt-in=kotlin.ExperimentalUnsignedTypes"
    }
}
```

See uniffi-rs docs [here](https://mozilla.github.io/uniffi-rs/kotlin/gradle.html#using-experimental-unsigned-types) and related merged PR [here](https://github.com/mozilla/uniffi-rs/pull/977) for more info.

I can't assign myself but @notmandatory feel free to assign me this one.

**Comment by artfuldev** [comment]
@notmandatory can we assign a version target for this and include it in the plan?

**Comment by artfuldev** [comment]
@thunderbiscuit The PR https://github.com/mozilla/uniffi-rs/pull/977 is not complete - as in there's still cases where the annotation isn't supported fully. It's referenced in the documentation as well as not complete:

> Uniffi is trying to add these annotations where necessary, but currently misses some places, see [PR #977](https://github.com/mozilla/uniffi-rs/pull/977) for details.

Along with this change we need to ensure our project compiles and runs on target platforms and kotlin versions and add a note to the docs on how to deal with this if someone faces an issue. Given we're compiling it and we might need to target lower versions of kotlin than consumers (as they don't have to compile it from source), it might make sense to leave things as they are. This isn't a new issue we didn't already know about, and we had some issues with some warnings showing up on compilation which we wanted to avoid.

I'd suggest coming back to this when all the cases are covered and we don't have to deal with this on our end anymore, or until we decide to support only kotlin versions >1.5.

**Comment by artfuldev** [comment]
@thunderbiscuit I missed the part about us using a later kotlin version than 1.5 🤦

**Comment by thunderbiscuit** [comment]
Yes well to be honest I'm not sure it's _enforced_ anywhere, but I think the following block will throw warnings at you if you're using a different version:

```kotlin
buildscript {
    ext.kotlin_version = '1.6.10'
    repositories {
        google()
        mavenCentral()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:7.0.4'
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
    }
}
```

**Comment by notmandatory** [comment]
I think it's reasonable to update the README as part of the PR to require Kotlin version `1.6.x`.

**Comment by notmandatory** [comment]
Fixed by #17
